src/rest.rs

```diff
@@ -1,22 +1,20 @@
 use crate::error::BrigadeError;
-use hyper::Method;
-use reqwest::Response;
-use serde::{Deserialize, Serialize};
-use serde_with::*;
+use reqwest::{IntoUrl, Method, RequestBuilder};
 
 #[derive(Debug, Clone)]
 pub struct ClientConfig {
     pub allow_insecure_connections: bool,
 }
 
 impl ClientConfig {
-    pub fn default() -> Self {
+    pub fn new() -> Self {
         Self {
             allow_insecure_connections: false,
         }
     }
 }
 
+#[derive(Debug, Clone)]
 pub struct Client {
     pub config: ClientConfig,
     pub address: String,
```

Comment:
I can't comment on the line below, but I would unexport the `rest` data member as you reexpose only the functionality you need within the `req` function